components/SignUpForm.tsx

```diff
@@ -0,0 +1,118 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import { useRef, useState } from "react";
+
+// Logo Import
+import logo from "../public/Cacart_logo.png";
+import googlelogo from "../public/google_logo.png";
+
+type FormInputs = {
+  email: string;
+  password: string;
+};
+
+export default function SignUpForm() {
+  // Set userData
+  const userData = useRef<FormInputs>({
+    email: "",
+    password: "",
+  });
+
+  const [error, setError] = useState("");
+
+  //   Value update handler
+  const valueUpdateHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    userData.current = { ...userData.current, [name]: value };
+  };
+
+  //   Form submit handler
+  const formSubmitHandler = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { email, password } = userData.current;
+
+    console.log(userData.current);
+  };
+
+  //   Error handle
```

Comment:
Wondering if this should be above our error constant?

Reply:
In the latest commit, I restructured the code to improve readability. The code now follows a consistent order:
1. Declaration of constants
2. Function implementations

Additionally, comments have been added where deemed necessary. Thank you for your suggestion.